awx/main/utils/ansible.py

```diff
@@ -44,13 +45,20 @@ def could_be_playbook(project_path, dir_path, filename):
     # show up.
     matched = False
     try:
-        for n, line in enumerate(open(playbook_path)):
+        for n, line in enumerate(codecs.open(
+            playbook_path,
+            'r',
+            encoding='utf-8',
+            errors='ignore'
+        )):
             if valid_playbook_re.match(line):
                 matched = True
+                break
```

Comment:
If we find a match on one line, there's no need to read the _entire_ file; we can bail out at this point.